src/_utils.js

```diff
@@ -622,13 +622,45 @@ export const booleanOption = opts => {
 }
 
 export const createReactComponentImportDeclaration = state => {
-  addDefault(
+  const uid = state.file.path.scope.generateUidIdentifierBasedOnNode(
+    state.file.path.node.id
+  )
+  const importIdentifierName = '_JSXImport' + uid.name
+
+  const importIdentifier = addDefault(
     state.file.path,
     typeof state.opts.styleModule === 'string'
       ? state.opts.styleModule
       : 'styled-jsx/style',
-    { nameHint: STYLE_COMPONENT}
+    { nameHint: importIdentifierName }
   )
+  // const _JSXStyle = _JSXImport<uid>;
```

Comment:
If we don't want to fix this properly (see https://github.com/vercel/styled-jsx/pull/716#issuecomment-847161924 and https://github.com/vercel/styled-jsx/pull/716#issuecomment-847670031) I think that we can get rid of addDefault from @babel/helper-module-imports and create the default import manually and name it `STYLE_COMPONENT` as the following is redundant

```js
import _JSXImport_ref from 'styled-jsx/style';
const _JSXStyle = _JSXImport_ref;
```

We could have:

```js
import _JSXStyle from 'styled-jsx/style';
```

Reply:
I do hope we can fix it properly here...
I tried to follow the hint "we will have to generate the identifier (the name of the import specifier)" but I'm not super clear what you mean and finally come up with this. Can you hint more if there's approach to generate the more unique but consistent (not change among compiles) identifier in `Program` enter phase?

Reply:
I didn't get it first, thought the generated uid id `ref` are all sames in those cases..now moving import id creation to program enter phase, also update a test case for it. Does it match what you said? @giuseppeg